js/apps/admin-ui/src/clients/authorization/policy/NameDescription.tsx

```diff
@@ -1,76 +1,29 @@
 import { useTranslation } from "react-i18next";
-import { useFormContext } from "react-hook-form";
-import { FormGroup, ValidatedOptions } from "@patternfly/react-core";
-
-import { HelpItem } from "ui-shared";
-import { KeycloakTextInput } from "../../../components/keycloak-text-input/KeycloakTextInput";
-import { KeycloakTextArea } from "../../../components/keycloak-text-area/KeycloakTextArea";
+import { TextAreaControl, TextControl } from "ui-shared";
 
 type NameDescriptionProps = {
-  prefix: string;
   isDisabled: boolean;
 };
 
-export const NameDescription = ({
-  prefix,
-  isDisabled,
-}: NameDescriptionProps) => {
+export const NameDescription = ({ isDisabled }: NameDescriptionProps) => {
   const { t } = useTranslation();
-  const {
-    register,
-    formState: { errors },
-  } = useFormContext();
 
   return (
     <>
-      <FormGroup
+      <TextControl
+        name="name"
         label={t("name")}
-        fieldId="kc-name"
-        helperTextInvalid={t("required")}
-        validated={
-          errors.name ? ValidatedOptions.error : ValidatedOptions.default
-        }
-        isRequired
-        labelIcon={
-          <HelpItem helpText={t(`${prefix}-nameHelp`)} fieldLabelId="name" />
-        }
-      >
-        <KeycloakTextInput
-          isDisabled={isDisabled}
-          id="kc-name"
-          data-testid="name"
-          validated={
-            errors.name ? ValidatedOptions.error : ValidatedOptions.default
-          }
-          {...register("name", { required: true })}
-        />
-      </FormGroup>
-      <FormGroup
+        rules={{ required: t("required") }}
+        isDisabled={isDisabled}
+      />
+      <TextAreaControl
+        name="description"
         label={t("description")}
-        fieldId="kc-description"
-        labelIcon={
-          <HelpItem
-            helpText={t(`${prefix}-descriptionHelp`)}
```

Comment:
I've removed this help text, as no label with the key `policy-descriptionHelp` exists.